diff --git a/dist/index.cjs b/dist/index.cjs
index 24eb9b391793067a216ce8577b297b2313f8d150..d0b44a29eec819247b22ce4c726fd9ef6da9765d 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -140,16 +140,20 @@ function domInject(inject = "body-last") {
       return "body.insertBefore(svgDom, body.lastChild);";
   }
 }
+const svgFilsStatsMap = new Map();
 async function compilerIcons(cache, svgOptions, options) {
   const { iconDirs } = options;
   let insertHtml = "";
   const idSet = /* @__PURE__ */ new Set();
   for (const dir of iconDirs) {
-    const svgFilsStats = fg__default.sync("**/*.svg", {
+    const svgFilsStats = svgFilsStatsMap.get(dir) || fg__default.sync("**/*.svg", {
       cwd: dir,
       stats: true,
       absolute: true
     });
+    if (!svgFilsStatsMap.has(dir)) {
+      svgFilsStatsMap.set(dir, svgFilsStats);
+    }
     for (const entry of svgFilsStats) {
       const { path: path2, stats: { mtimeMs } = {} } = entry;
       const cacheStat = cache.get(path2);
diff --git a/dist/index.mjs b/dist/index.mjs
index e160cd1d574b30cace7825be7c4e967558b3f6aa..4f15c5e65283da08ad851e005d2589884e723a3b 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -126,16 +126,20 @@ function domInject(inject = "body-last") {
       return "body.insertBefore(svgDom, body.lastChild);";
   }
 }
+const svgFilsStatsMap = new Map();
 async function compilerIcons(cache, svgOptions, options) {
   const { iconDirs } = options;
   let insertHtml = "";
   const idSet = /* @__PURE__ */ new Set();
   for (const dir of iconDirs) {
-    const svgFilsStats = fg.sync("**/*.svg", {
+    const svgFilsStats = svgFilsStatsMap.get(dir) || fg.sync("**/*.svg", {
       cwd: dir,
       stats: true,
       absolute: true
     });
+    if (!svgFilsStatsMap.has(dir)) {
+      svgFilsStatsMap.set(dir, svgFilsStats);
+    }
     for (const entry of svgFilsStats) {
       const { path: path2, stats: { mtimeMs } = {} } = entry;
       const cacheStat = cache.get(path2);
